### ISSUE TITLE
`no-unused-vars` false positive

### ISSUE THREAD

**Issue by eventualbuddha** [oxc-project/oxc#5391]
This line triggers a `no-unused-vars` warning when it should not: https://github.com/votingworks/vxsuite/blob/720f72ac404bfff011034f275317a02f5d69cff9/libs/ui/src/hand_marked_paper_ballot_prose.tsx#L10:

Here's a stripped down–but not minimal–version of the code that still reproduces the issue:

```ts
import styled from 'styled-components';

import { Prose, ProseProps } from './prose';

interface Props extends ProseProps {
  density?: number;
}

export const HandMarkedPaperBallotProse = styled(Prose)<Props>`
  line-height: ${({ density }) => (density !== 0 ? '1.1' : '1.3')};
`;
```

Here's the output from `oxlint`:

```
❯ npx oxlint@latest libs/ui/src/hand_marked_paper_ballot_prose_minimal.tsx

  ⚠ eslint(no-unused-vars): Parameter 'density' is declared but never used.
    ╭─[libs/ui/src/hand_marked_paper_ballot_prose_minimal.tsx:10:21]
  9 │ export const HandMarkedPaperBallotProse = styled(Prose)<Props>`
 10 │   line-height: ${({ density }) => (density !== 0 ? '1.1' : '1.3')};
    ·                     ───┬───
    ·                        ╰── 'density' is declared here
 11 │ `;
    ╰────
  help: Consider removing this parameter.

Finished in 4ms on 1 file with 93 rules using 3 threads.
Found 1 warning and 0 errors.
```

**Comment by camchenry** [comment]
I found I was able to reproduce this using `npx oxlint@latest` against the given code example. However, when I put it into our Rust test suite, I didn't see the same false positive.

@camc314 @DonIsaac I'd be happy to try and look into this if I have some time, but do you know why it might be triggering a warning with `npx oxlint`, but not when I add this as a test case?

**Comment by DonIsaac** [comment]
Lemme take a look and repro in my end.

**Comment by DonIsaac** [comment]
I can reproduce this in the playground but not in local test cases. The problem is related to the interface heritage clause. When removed or moved below the tagged template function, `density` has a single read reference. In the snippet above, `density` has no references at all.

I guess this semantic bug was fixed in a recent PR, so this should be fixed in the next release. Nonetheless, I'll make a PR adding this as a test case.
